HWs/HW5/Routers/Routers/ConfigurationGenerator.cs

```diff
@@ -0,0 +1,151 @@
+namespace ConfigurationGenerator;
+
+using System.Text;
+using System.Text.RegularExpressions;
+using UniqueList;
+
+/// <summary>
+/// A class for building the optimal configuration of routers, which is the maximum spanning tree.
+/// </summary>
+public static class ConfigurationGenerator
+{
+    /// <summary>
+    /// Generation of the maximum spanning tree by the Prim's algorithm.
+    /// </summary>
+    /// <param name="network"> Initial network. </param>
+    /// <returns> Optimal network. </returns>
+    public static string CreateOptimalConfiguration(string network)
+    {
+        if (network == string.Empty)
+        {
+            throw new ArgumentException("Network can't be empty.");
+        }
+
+        EdgeOfTheNetwork[]? edges = null;
+        int routersCount = 0;
+
+        try
+        {
+            (edges, routersCount) = ParseNetwork(network);
+        }
+        catch (ArgumentException)
+        {
+            throw;
+        }
+
+        var configuration = new List<EdgeOfTheNetwork>();
+        var connectedRouters = new UniqueList();
+        connectedRouters.Add(1);
+
+        while (connectedRouters.Count < routersCount)
+        {
+            EdgeOfTheNetwork currentMaximumEdge = CurrentMaximumEdge(edges, connectedRouters);
+
+            if (currentMaximumEdge.Bandwidth == -1)
+            {
+                throw new ArgumentException("The network is disconnected.");
+            }
+
+            connectedRouters.Add(currentMaximumEdge.FirstRouter);
+            connectedRouters.Add(currentMaximumEdge.SecondRouter);
+
+            configuration.Add(currentMaximumEdge);
+        }
+
+        return BuildConfigurationForFile(configuration, routersCount);
+    }
+
+    private class EdgeOfTheNetwork
+    {
+        internal EdgeOfTheNetwork(int firstRouter, int secondRouter, int bandwidth)
+        {
+            this.FirstRouter = firstRouter;
+            this.SecondRouter = secondRouter;
+            this.Bandwidth = bandwidth;
+        }
+
+        internal int FirstRouter { get; }
+        internal int SecondRouter { get; }
+        internal int Bandwidth { get; }
+    }
+
+    private static EdgeOfTheNetwork CurrentMaximumEdge(EdgeOfTheNetwork[] edges, UniqueList connectedRouters)
+    {
+        var result = new EdgeOfTheNetwork(0, 0, -1);
```

Comment:
```suggestion
    private static EdgeOfTheNetwork? CurrentMaximumEdge(EdgeOfTheNetwork[] edges, UniqueList connectedRouters)
    {
        EdgeOfTheNetwork? result = null;
```
В твоем варианте два недостатка:
* используется магическая константа `-1`,
* создается несуществующее ребро `EdgeOfTheNetwork(0, 0, -1)`.

Так более наглядно: `null`, если ребра нет.
Всё равно понадобится переменная для поиска минимума: её стоит инициализировать значением `int.MinValue`, а не `-1`.